File: utils/cpp/link_list.cpp
#include <vector>
using namespace std;

struct ListNode {
  int val;
  ListNode* next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode* next) : val(x), next(next) {}
};

ListNode* array2list(vector<int>&& arr) {
  ListNode* ret;
  for (auto item : arr) {
    ret = new ListNode(item, ret);
  }
  return ret;
}

File: utils/go/link_list.go
package utils

type ListNode struct {
	Val  int
	Next *ListNode
}

func array2List(arr ...int) *ListNode {
	head := &ListNode{
		Val: arr[0],
	}

	curr := head
	for _, val := range arr[1:] {
		curr.Next = &ListNode{
			Val: val,
		}
		curr = curr.Next
	}

	return head
}

func list2Array(head *ListNode) []int {
	arr := make([]int, 0)
	for head != nil {
		arr = append(arr, head.Val)
		head = head.Next
	}
	return arr
}
